### ISSUE TITLE
I have a question about Usage.

### ISSUE THREAD

**Issue by star4s** [nickyisadog/latent-diffusion-inpainting#8]
You mentioned to combine the autoencoder with the dissusion model in the second step.
When I check combine.ipynb in your code, there are two parameter, ldm_model_path and auto_model_path.

I think that auto_model_path came from the first step, Finetune the autoencoder, and ldm_model_path came from third step, Finetune Latent diffusion model.

So, The process step is 1 -> 3 -> 2 -> 4 in the Usage.
Is it correct?

If the step of your Usage is correct, how can I get  ldm model for ldm_model_path in the second step,  combine the autoencoder with the dissusion model.

Thank you for your attention.

**Comment by nickyisadog** [comment]
1 - > 2 -> 3 -> 4

The whole pipeline consist of two model, the auto-encoder and the diffusion model.

For inference,
First, the encoder of the auto-encoder compress the images into smaller dimension(latent vector).
Then, the diffusion model (diffusion process) will take the latent vector as an input and generate a new latent vector.
In the end, the decoder of the auto-encoder decode the newly generated latent vector back to image.


For training,
Why we want to fine-tune the auto-encoder?? we want our images to encode and decode better(try to retain more information).
Why we want to fine-tune the diffusion model?? To learn what to generate.
The model (last.ckpt from the original repo) contain both the auto-encoder and diffusion model.
Therefore, combine.ipynb is replacing the auto-encoder in last.ckpt with your fine-tuned auto-encoder in step 1 

Therefore,
1 -> 3 -> 2 -> 4  is INCORRECT 
since **the training of a diffusion model depends on a fixed auto-encoder**
In step 1, you get a new auto encoder.
In step 3, you finetune the last.ckpt with an old auto-encoder
In step 2, you replace the old auto-encoder with new auto-encoder in the last.ckpt. 
Now,  your new last.ckpt,  the diffusion model (trained base on an old auto-encoder) might not work well with a new auto-encoder.


I think for most of the cases, you can just start from step 3. You only want to do 1 and 2 when u have very unusual images like me.

**Comment by star4s** [comment]
@nickyisadog Thank you for your quick reply.

Since you deeply explain about the pip line, I can understand, well.

1. Your mean is that ldm_model_path is ldm/models/ldm/inpainting_big/lask.ckpt in step 2, isn't it?
ldm_model_path's model was downloading at https://heibox.uni-heidelberg.de/f/4d9ac7ea40c64582b7c9/?dl=1.
Could you check that fact? 


I have one more question about your train data

2. In repo, there are 5 pair files in /Kvasir-SEG/images and /Kvasir-SEG/masks. 
How many do you use the data for training?
Do you use just the 5 pair files?

3. In addition, Do you use the same dataset for 1st step and  3rd step?
 
Could you check about 3 points ?

Thank you for your attention.

**Comment by nickyisadog** [comment]
1. yes
2. No, i use around 5000 images but the format is the same as in Kvasir-SEG
3. You only need images to do the 1st step and you need both images and masks to do 3rd step

**Comment by star4s** [comment]
@nickyisadog Thank you for your quick reply.
I can keep going my work for your help.
i appreciate.

**Comment by dazmashaly** [comment]
> 1. ldm_model_path

hello, can you help, from what i understand the ldm_model_path is the last.ckpt from you get from finetuneing the autoencode, but then what is the auto_model_path? i tried using the original model but i get missmatched keys
